library/src/test/java/com/patloew/colocation/CoLocationTest.kt

```diff
@@ -89,6 +86,52 @@ class CoLocationTest {
         )
     }
 
+    @ParameterizedTest
+    @ValueSource(ints = [
+        LocationRequest.PRIORITY_HIGH_ACCURACY,
+        LocationRequest.PRIORITY_BALANCED_POWER_ACCURACY,
+        LocationRequest.PRIORITY_LOW_POWER,
+        LocationRequest.PRIORITY_NO_POWER
+    ])
+    fun getCurrentLocation(priority: Int) {
+        val location = mockk<Location>()
+        testTaskWithCancelReturns(
+            createTask = { locationProvider.getCurrentLocation(priority, any()) },
+            taskResult = location,
+            expectedResult = location,
+            expectedErrorException = TestException(),
+            cancelResult = null,
+            coLocationCall = { coLocation.getCurrentLocation(priority) }
+        )
+    }
+
+    @ParameterizedTest
+    @ValueSource(ints = [
+        LocationRequest.PRIORITY_HIGH_ACCURACY,
+        LocationRequest.PRIORITY_BALANCED_POWER_ACCURACY,
+        LocationRequest.PRIORITY_LOW_POWER,
+        LocationRequest.PRIORITY_NO_POWER
+    ])
+    fun `cancelling getCurrentLocation cancels task`(priority: Int) {
+        lateinit var token: CancellationToken
+
+        every {
+            locationProvider.getCurrentLocation(priority, any())
+        } answers {
+            token = arg(1)
+            mockk(relaxed = true)
+        }
+
+        val deferred = testCoroutineScope.async(start = CoroutineStart.UNDISPATCHED) {
+            coLocation.getCurrentLocation(priority)
+        }
+
+        deferred.cancel()
+
+        assertTrue(deferred.isCancelled)
+        assertTrue(token.isCancellationRequested)
```

Comment:
`slot()` could be used to capture the argument.
```suggestion
        val token = slot<CancellationToken>()
        every { locationProvider.getCurrentLocation(priority, capture(token)) } returns mockk(relaxed = true)

        val deferred = testCoroutineScope.async(start = CoroutineStart.UNDISPATCHED) {
            coLocation.getCurrentLocation(priority)
        }

        deferred.cancel()

        assertTrue(deferred.isCancelled)
        assertTrue(token.captured.isCancellationRequested)
```

Reply:
Update almost as suggested, although I named the slot `tokenSlot`, since it isn't directly a token itself.